api/api.go

```diff
@@ -120,42 +162,24 @@ func NewAPIWithVersion(ctx context.Context, config *conf.GlobalConfiguration, db
 		AllowCredentials: true,
 	})
 
-	api.handler = corsHandler.Handler(mux)
+	api.handler = corsHandler.Handler(chi.ServerBaseContext(r, ctx))
 
 	return api
 }
 
-func (a *API) logCompleted(ctx context.Context, wp mutil.WriterProxy, r *http.Request) {
-	log := gcontext.GetLogger(ctx).WithField("status", wp.Status())
-
-	start := gcontext.GetStartTime(ctx)
-	if start != nil {
-		log = log.WithField("duration", time.Since(*start))
-	}
-
-	log.Infof("Completed request %s. path: %s, method: %s, status: %d", gcontext.GetRequestID(ctx), r.URL.Path, r.Method, wp.Status())
-}
-
-func (a *API) populateContext(ctx context.Context, w http.ResponseWriter, r *http.Request) context.Context {
+func requestIDCtx(w http.ResponseWriter, r *http.Request) (context.Context, error) {
 	id := uuid.NewRandom().String()
-	log := a.log.WithField("request_id", id)
-
-	log = log.WithFields(logrus.Fields{
-		"method": r.Method,
-		"path":   r.URL.Path,
-	})
-
+	ctx := r.Context()
 	ctx = gcontext.WithRequestID(ctx, id)
-	ctx = gcontext.WithLogger(ctx, log)
-	ctx = gcontext.WithStartTime(ctx, time.Now())
+	return ctx, nil
+}
 
+func (a *API) instanceConfigCtx(w http.ResponseWriter, r *http.Request) (context.Context, error) {
+	ctx := r.Context()
 	if gcontext.GetPaymentProviders(ctx) == nil {
```

Comment:
Also - wouldn't we want to inject the db instance here? It lives on the API itself, it seems like this should do the same. 

It will need to be changed for multitenancy, maybe, but I am not sure. Mostly I'd rather not conflate the issues.